core/src/main/java/org/aerogear/mobile/core/reactive/CallableRequest.java

```diff
@@ -0,0 +1,98 @@
+package org.aerogear.mobile.core.reactive;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+
+import android.util.Log;
+
+import org.aerogear.mobile.core.Request;
+import org.aerogear.mobile.core.Responder;
+import org.aerogear.mobile.core.utils.SanityCheck;
+
+/**
+ * This class will synchronously invoke a callable parameter when
+ * {@link CallableRequest#respondWith(Responder)} is invoked
+ *
+ * @param <T> a type that this request generates
+ */
+public final class CallableRequest<T> extends AbstractRequest<T> {
+
+    private static final String ERROR_TAG = "UNCAUGHT_EXCEPTION";
+    private final Callable<T> callable;
+    private Thread callableThread = null;
+
+    public CallableRequest(Callable<T> callable) {
+        SanityCheck.nonNull(callable, "callable");
+        this.callable = callable;
+    }
+
+    @Override
+    public Request<T> respondWithActual(AtomicReference<Responder<T>> responderRef) {
+
+        if (responderRef.get() == null) { // responder may have been disconnected.
+            return this;
+        }
+
+        T value = null;
+        Exception exception = null;
+
+        /*
+         * The reason we calculate the exception and value parameters in one try block and then call
+         * the responder in another is to isolate the calculation of the result from the handling of
+         * the response.
+         *
+         * Otherwise it would be possible for the call to succeed but then the responder would throw
+         * an exception in the onSuccess method that would be passed to the responder's onException.
+         *
+         */
+        try {
+            callableThread = Thread.currentThread();
+            value = callable.call();
+            synchronized (callable) { // We are synchronizing on the callable because we don't want
+                                      // the thread reference to go null in the cancel method.
+                callableThread = null;
+            }
+        } catch (Exception e) {
+            exception = e;
+        }
+
+        Responder<T> responder = responderRef.get();
+        if (responder == null) { // responder may have been disconnected while the calculation was
+                                 // performed.
+            return this;
+        }
+
+        try {
+            if (exception == null) {
+                responder.onResult(value);
```

Comment:
if `responder` have been disconnected after the previous check, but before this call we will get a `NullPointerException`
I think we should synchronize this block

Reply:
The local repsonder reference can't be set to null by an external thread.  The AtomicReference may lose its reference, but we don't care about it anymore.